saleor/graphql/channel/mutations.py

```diff
@@ -125,9 +126,16 @@ def _save_m2m(cls, info, instance, cleaned_data):
         remove_shipping_zones = cleaned_data.get("remove_shipping_zones")
         if remove_shipping_zones:
             instance.shipping_zones.remove(*remove_shipping_zones)
-            instance.shipping_method_listings.filter(
+            shipping_channel_listings = instance.shipping_method_listings.filter(
                 shipping_method__shipping_zone__in=remove_shipping_zones
-            ).delete()
+            )
+            shipping_method_ids = list(
+                shipping_channel_listings.values_list("shipping_method_id", flat=True)
+            )
+            shipping_channel_listings.delete()
+            drop_invalid_shipping_methods_relations_for_given_channels.delay(
```

Comment:
Are we sure that should be a task? 
What if I remove the shipping method from channel `X` and a few seconds after(before the task is executed)  some client tries to complete checkout with this shipping method assigned to checkout?

Reply:
Good point, maybe it shouldn't be a task.